dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/datastore/DataStoreFileHandler.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.datastore
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.storage.datastore.DataStoreCallback
+import com.datadog.android.api.storage.datastore.DataStoreHandler
+import com.datadog.android.core.internal.persistence.Deserializer
+import com.datadog.android.core.internal.utils.submitSafe
+import com.datadog.android.core.persistence.Serializer
+import java.util.concurrent.ExecutorService
+
+internal class DataStoreFileHandler(
```

Comment:
should we have any internal method (not exposed via public API) to remove all the content of data store? Given that we have `Datadog.clearAllData` method, should it clear also the data store?

Reply:
On ios there is no such mechanism, but we should have one if there is a usecase. Is clearAllData supposed to clear only data that we would send to the backend, or all sdk data? From the documentation in the interface it implies that it clears only data that would be sent, and at the moment the DS data is not sent

Reply:
It depends on the usecase indeed. I think it is rather not about preventing data to be sent, but just to clear all the data written if customer wants to do it because of the different privacy/legal concerns.

Reply:
I've opened a task to do this in a separate pr